README.md

```diff
@@ -1,102 +1,262 @@
+- [`create_logout_request_url`](#create_logout_request_url)- Creates a request URL to log out the user.
+- [`create_logout_response_url`](#create_logout_response_url) - Creates a URL to confirm a successful logout.
+- [`create_metadata`](#create_metadata) - Returns the XML metadata used during the initial SAML configuration.
 
-```coffee
-  # -- REQUIRED --
-  # Returns a redirect URL, at which a user can login
-  sp.create_login_url(idp, cb)
+#### Example
+```javascript
 
-  # Returns user object, if the login attempt was valid.
-  sp.assert(idp, request_body, cb)
+  // Initialize options object
+  var sp_options = {
+    entity_id: "https://sp.example.com/metadata.xml",
+    private_key: fs.readFileSync("key-file.pem").toString(),
+    certificate: fs.readFileSync("cert-file.crt").toString(),
+    assert_endpoint: "https://sp.example.com/assert",
+    force_authn: true,
+    auth_context: { comparison: "exact", class_refs: ["urn:oasis:names:tc:SAML:1.0:am:password"] },
+    nameid_format: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+    sign_get_request: false,
+    allow_unencrypted_assertion: true
+  }
 
-  # -- OPTIONAL --
-  # Returns a redirect URL, at which a user is logged out.
-  sp.create_logout_url(idp, cb)
+  // Call service provider constructor with options
+  var sp = saml2.ServiceProvider(sp_options);
+
+  // Example use of service provider.
+  // Call metadata to get XML metatadata used in configuration.
+  var metadata = sp.create_metadata();
 
-  # Returns XML containing service-provider parameters.
-  # For use during initial SAML configuration
-  sp.create_metadata(idp, cb)
 ```
 
-## Helper Methods
+#### Service provider function definitions
 
-We will break each of the `service_provider` methods into minimal, testable methods.
+<a name="create_login_request_url" />
+##### create_login_request_url(IdP, options, cb)
+Get a URL to initiate a login.
 
-```coffee
-  ... TODO ...
-  parse_xml
-  parse_assert
-  createAuthRequest
-```
+Takes the following arguments:
+- `IdP` - [IdP](#IdentityProvider)
+- `options`
+  - `relay_state` - SAML relay state.
+  - `auth_context` - Specifies `AuthnContextClassRef`.  This can also be configured on the [SP](#ServiceProvider).
+  - `nameid_format` - Format for Name ID.  This can also be configured on a per-method basis.
+  - `force_authn` - Boolean that allows forcing re-authentication of users even if the user has a SSO session with the IdP.  This can also be configured on the [IdP](#IdentityProvider) or [SP](#ServiceProvider).
+  - `sign_get_request` - Boolean, if true signs the request.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `cb(error, url, request_id)` - Callback called with the login URL and ID of the request.
+
+
+<a name="redirect_assert" />
+##### redirect_assert(IdP, options, cb)
+Gets a user object if the login attempt is valid, inflates the response before parsing it.
```

Comment:
Same as above - this description doesn't explain the function to me. Where does it get the user from? Why is it called `redirect_assert`? What is inflating?